### DISCUSSION TITLE
Keep workflow artifacts on reruns

### DISCUSSION THREAD

**Discussion by singhprd** [community/community#17854] [Actions]
Currently when a workflow is re-run the artifacts for the initial run are overwritten (even if they're uniquely named).

### Request:

- Provide an option to persist workflow artifacts between re-runs

### Impact:

This will help me when debugging flakey runs, which pass on re-run. At the moment the debugging information in the artifacts is lost forever.

---

#### Context

GitHub support have confirmed the behaviour to me that artifacts are overwritten on re-run, but there is a nuance with partial re-runs:

> A run only has one set of artifacts. Any time a user presses "re-run all jobs" the original artifacts will be removed and replaced with any artifacts from the new attempt. This has always been the case.
> 
> "re-run failed jobs" and "re-run a specific job" are slightly different. If you re-run a subset of jobs, we do not delete the artifacts as jobs can share artifacts. They are simply moved into the new attempt of the run. See https://docs.github.com/en/actions/managing-workflow-runs/re-running-workflows-and-jobs.
>

**Comment by ahasna** [comment]
Is there any update on this?

**Comment by dcyruschow-cat** [comment]
> "re-run failed jobs" and "re-run a specific job" are slightly different. If you re-run a subset of jobs, we do not delete the artifacts as jobs can share artifacts. They are simply moved into the new attempt of the run. See https://docs.github.com/en/actions/managing-workflow-runs/re-running-workflows-and-jobs.

What I'm noticing with my private repo job (sorry I cannot share) is that viewing previous run will still show the old artifacts from the partially succeeded subset of jobs (so I find "moved into the new attempt of the run" misleading/incorrect), whilst the re-run only shows the artifacts from those jobs that were indeed re-run.
